packages/core-compat-api/src/components/SystemIcon.tsx

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useApp } from '@backstage/core-plugin-api';
+import React from 'react';
+import { compatWrapper } from '../compatWrapper';
+
+/**
+ * @public
+ * Props for the System Icon component.
+ */
+export type SystemIconProps = {
+  // The id of the system icon to render.
+  id: string;
+  // An optional fallback element to render when the system icon is not found.
+  fallback?: JSX.Element;
```

Comment:
Wanna hear your thoughts on this element approach vs `IconComponent` 😁

Could be the case that we want the `SystemIcon` to accept a `fontSize` prop to set the size, I think that's fairly common. Might be that the best approach for that is to always set an outer size and let the icon inherit it though.

Reply:
I forgot about this `IconComponent` 😄 . I guess we can accept all `IconComponentProps` and forward them to the returned system component or fallback... I'll try it. Thanks!

Reply:
https://github.com/backstage/backstage/pull/22892/commits/d5f859bd8749c71f0d6fa03220edd4007ba0ac62